src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -328,7 +329,10 @@ private TagBuilder GenerateTextBox(ModelExplorer modelExplorer, string inputType
             {
                 format = GetFormat(modelExplorer, inputTypeHint, inputType);
             }
-
+            if (modelExplorer.Model != null && string.Equals("week", inputType, StringComparison.OrdinalIgnoreCase))
```

Comment:
Like the general approach.

This isn't correct when the user uses `asp-format`. Should be inside the `string.IsNullOrEmpty(format)` block.

Should also probably be done _instead of_ the `GetFormat()` call. Right now, this comes up with a default `datetime-local` format that isn't used. That might look like
``` c#
if (string.IsNullOrEmpty(format))
{
    if (!metadata.HasNonDefaultEditFormat &&
        string.Equals("week", inputType, StringComparison.OrdinalIgnoreCase) &&
        (modelExplorer.Model is DateTime || modelExplorer is DateTimeOffset))
    {
        modelExplorer = modelExplorer.GetExplorerForModel(FormatWeekHelper.GetFormattedWeek(modelExplorer));
    }
    else
    {
        format = GetFormat(modelExplorer, inputTypeHint, inputType);
    }
}
```
May want to remove `metadata.HasNonDefaultEditFormat` check inside `GetFormattedWeek(...)`.